File: LibWarDB2/WDC3.cs
using LibWarDB2.Structures;

namespace LibWarDB2
{
    public class WDC3
    {
        public WDC3Header Header { get; set; }
        public WDC3SectionHeader[] SectionHeaders { get; set; }
        public FieldStructure[] Fields { get; set; }
        public FieldStorageInfo[] FieldInfo { get; set; }
        public byte[] PalletData { get; set; }
        public byte[] CommonDate { get; set; }
        public Section[] DataSections { get; set; }
    }
}

File: LibWarDB2VB/WDC3.vb
Imports System.IO
Imports LibWarDB2VB.LibWarDB2.Extensions
Imports LibWarDB2VB.LibWarDB2.Structures

Namespace LibWarDB2
    Public Class WDC3
        Public Const Signature As UInteger = 860046423
        Public Property Header As WDC3Header
        Public Property SectionHeaders As WDC3SectionHeader()
        Public Property Fields As FieldStructure()
        Public Property FieldInfo As FieldStorageInfo()
        Public Property PalletData As Byte()
        Public Property CommonData As Byte()
        Public Property DataSections As Section()

        Public Sub Load(ByVal path As String)
            Using reader = New BinaryReader(File.Open(path, FileMode.Open, FileAccess.Read))
                Dim wdc = reader.ReadWDC3()
                Header = wdc.Header
                SectionHeaders = wdc.SectionHeaders
                Fields = wdc.Fields
                FieldInfo = wdc.FieldInfo
                PalletData = wdc.PalletData
                CommonData = wdc.CommonData
                DataSections = wdc.DataSections
            End Using
        End Sub

        Public Sub Save(ByVal path As String)
            Using writer = New BinaryWriter(File.Open(path, FileMode.Create, FileAccess.Write))
                writer.Write(Me)
            End Using
        End Sub
    End Class
End Namespace
